application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadAutoArchiver.java

```diff
@@ -110,10 +110,20 @@ private void autoArchiveForThread(ThreadChannel threadChannel, Instant archiveAf
                 .setColor(HelpSystemHelper.AMBIENT_COLOR)
                 .build();
 
-            threadChannel.sendMessage(author.getAsMention())
-                .addEmbeds(embed)
-                .flatMap(any -> threadChannel.getManager().setArchived(true))
-                .queue();
+            Consumer<Member> messageAuthorOnArchive =
+                    author -> threadChannel.sendMessage(author.getAsMention())
+                        .addEmbeds(embed)
+                        .flatMap(any -> threadChannel.getManager().setArchived(true))
+                        .queue();
+
+            Consumer<Throwable> handleFailure =
+                    error -> logger.debug("was unable to retrieve member with id: {}",
+                            threadChannel.getOwnerIdLong(), error);
+
+            threadChannel.getGuild()
+                .retrieveMemberById(threadChannel.getOwnerIdLong())
+                .queue(messageAuthorOnArchive, handleFailure);
```

Comment:
works. but might as well just use `flatMap` instead of two `queue`s

Reply:
like
```java
threadChannel.getGuild()
  .retrieveMemberById(...)
  .flatMap(author -> threadChannel.sendMessage(...).addEmbeds(...))
  .flatMap(any -> threadChannel.getManager().setArchived(...))
  .queue();
  ```

Reply:
Should I handle the error part as well? Incase retrieve fails?

Reply:
I kinda went with what you suggested earlier but had some trouble chaining error logs, that's why changed it to this form.

Reply:
retrieving a member should always work. no need to handle it.

Reply:
(famous last words)